### ISSUE TITLE
Implement new page editor header & breadcrumbs

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8136]
[Design link](https://www.figma.com/file/h67EsVXdWsfu38WGGxWfpi/Wagtail-2022?node-id=3218%3A31572)

Reference screenshot:

![page-editor-header](https://user-images.githubusercontent.com/877585/158591835-19ca8a0a-49c3-48cf-8c03-91223dccf574.png)

Accessibility annotations (based on an outdated version of our August 2022 release’s design, _not_ current target):

![page-editor-header-accessibility-annotations](https://user-images.githubusercontent.com/877585/158606273-d5e90687-d0c5-405d-861e-4a21858f6783.png)

## Tasks

- Create as a new component so existing existing headers elsewhere in Wagtail work as-is
- Implement new UI
- Implement breadcrumb within the header
- Add Storybook stories as needed
- Review impact on existing customisations and implement if needed: `breadcrumb`, `right_column`, `extra_rows`.
- Excluded: tabs (used to be in the header, now outside)

## Acceptance criteria

- [ ] Notifications behave the same as currently and push the header downwards
- [ ] Matches design 

## Testing / review plan

- UI testing in German by @stevedya 
- Accessibility tests by @wagtail/accessibility
- Code review by @thibaudcolas

**Comment by phildexter** [comment]
@kaedroho are you ok to help @stevedya with some of the failing Python tests on this before @thibaudcolas reviews?

**Comment by stevedya** [comment]
@kaedroho This PR is where I would need some help #8231 pleaaaasee 🙏 . A lot of the elements that the tests rely on have been moved or removed for this design to work

**Comment by thibaudcolas** [comment]
Merged!
